### DISCUSSION TITLE
Best way to set Environment Variables

### DISCUSSION THREAD

**Discussion by michaelweinold** [jupyterlite/jupyterlite#1151] [Q&A]
As I understand it, setting environment variables applicable to all kernels started in the JupyterLite environment is currently not supported.

In the case of a regular Jupyter environment, environment variables [can be set by modifying the `kernel.json` file](https://stackoverflow.com/a/53595397/):

```json
{
 "display_name": "Python 2 with environment",
 "language": "python",
 "argv": [
  "/usr/bin/python2",
  "-m",
  "ipykernel_launcher",
  "-f",
  "{connection_file}"
 ],
 "env": {"LD_LIBRARY_PATH":""}
}
```

It seems that at present, the best workaround is to use:

```python
import os
os.environ["MYVAR"] = "1"
```

Or could I use one of the options listed here: ["Configuring `jupyter_lite_config.json`, `jupyter-lite.json` and `overrides.json`"](https://jupyterlite.readthedocs.io/en/stable/howto/configure/config_files.html#configuring-jupyter-lite-config-json-jupyter-lite-json-and-overrides-json)?

**Comment by jtpio** [comment]
> setting environment variables applicable to all kernels started in the JupyterLite environment is currently not supported.

Probably the main challenge with this would be that JupyterLite kernels run in a Web Worker, and not in a dedicated process like with a traditional Jupyter setup. So having something generic that works for all kernels could be tricky.

Maybe the best bet for now would be to run (via an third-party extension) code on kernel startup (https://github.com/jupyterlite/jupyterlite/issues/461)
